documentation/cwl-vs-wdl.md

```diff
@@ -1,24 +1,38 @@
+* The original CWL does not have an option for disk space, but the WDL does, as it is a soft requirement for running on Terra.  
+* The CWL generates config files using an InlineJavascriptRequirement, which is run before the CWL equivivalent of a task's command section begins. The WDL generates them using an inline Python script during the beginning of a task's command section.  
+* Occasionally, the WDLs need to relocalizes all files into the working directory which requires more disk space than the CWL would [due to this workaround](https://github.com/DataBiosphere/analysis_pipeline_WDL/issues/2). This workaround exists in vcf-to-gds' unique variant IDs task and in ld-pruning's merge task.
+* A few R scripts require chromosome numbers to be passed in on the command line rather than in the configuration file. In order to do this, chromosome number is written to a file (completely separate from the configuration file) in the task's inline Python section. Upon exiting the Python block, this extra file is read in BASH and then passed to the Rscript call as an argument (as opposed to being in the configuration file). Although the actual call to the R script is identical as the CWL also passes the chr number as an argument, the CWL is able to rely on its inline Javascript to do this, while the WDL must use this workaround to pass the chr number out of the Python scope.
 
-* The WDL relocalizes all files into the working directory in the unique variant IDs step. [This issue is explained in detail here](https://github.com/DataBiosphere/analysis_pipeline_WDL/issues/2).
+## "Not Strictly Necessary But In WDL This Makes More Sense"
+* The original CWL allocates memory and CPUs on an overall level, while the WDL does it on a task level. In other words, the WDL is more granular.  
+	* Reasoning: GCS has stricter requirements than AWS regarding storage. Doing it this way also allows resource-heavy tasks (such as check_gds in vcf-to-gds-wf) to only use the large number of resources they need in their respective task, instead of the whole workflow.  
 
-* The original CWL does not have an option for disk space, but the WDL does, as it is a soft requirement for running on Terra.  
+## Different Design Choices
+* The CWL imports other CWLs. The WDL does not import other WDLs, except in the case of the checker workflow.  
+	* Reasoning: It is possible for WDLs to contain just tasks and be imported into another task, but in some contexts it can be slightly less secure.
 
-* The CWL generates config files using an InlineJavascriptRequirement, which is run before the CWL equivivalent of a task's command section begins. The WDL generates them using an inline Python script during the beginning of a task's command section.  
+# ld-pruning-wf.wdl
+## Strictly Necessary Changes
+* Similiar to Workflow A's file relocalization trick in unique_variant_ids, this workflow's merge_gds task has to use the same workaround.
```

Comment:
I may be confused, but did we decide to get rid of the Workflow A terminology?

Reply:
Good spot, I missed that one.